geps/gep-1619.md

```diff
@@ -157,49 +281,401 @@ Generally, the implementation API programs the dataplane API; however these two
-- Session Cookie Configuration
-- Session Cookie Strategy
+Kubernetes provides an API that allows you to enable [session affinity](https://kubernetes.io/docs/reference/networking/virtual-ips/#session-affinity)
+on service objects. It ensures consistent sessions by utilizing the client's IP address and also offers the option to
+set a timeout for the maximum session duration. Implementations of Gateway API don't rely on the service IP, so this
+particular application of session affinity doesn't impact the design of the GEP. However, we recommend that users enable
+session affinity using the Kubernetes service object to for applications requiring persistence, particularly due to the
+fact that in-cluster traffic will make use of the service IP.
 
-**Session Cookie Configuration**
-
-Session cookie configuration are the parameters and attributes the dataplane uses to create a cookie for establishing a session.
-
-A set-cookie request can be broken down into [3 main components](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie):
-- Cookie Name
-- Cookie Value
-- Cookie Attributes
+## API
 
-Only the Cookie Name and Cookie Value are required. All Cookie Attributes are optional. Attributes are simply key=value pairs with the value as optional for some attributes.
+### Session Persistence API
 
-The set-cookies attributes defined by [rfc6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2) are:
+In this section, we will explore the questions and design elements associated with a session persistence API.
+
+#### GO
+
+```go
+// SessionPersistencePolicy provides a way to define session persistence rules
+// for a service or route.
+//
+// Support: Core
+type SessionPersistencePolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of SessionPersistencePolicy.
+    Spec SessionPersistencePolicySpec `json:"spec"`
+
+    // Status defines the current state of SessionPersistencePolicy.
+    Status SessionPersistencePolicyStatus `json:"status,omitempty"`
+}
+
+// SessionPersistencePolicySpec defines the desired state of
+// SessionPersistencePolicy.
+// Note: there is no Override or Default policy configuration.
+type SessionPersistencePolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // HTTPRoutes, GRPCRoute, Services are the only valid API target
+    // references.
+	//
+	// Services as target reference are core support level.
+	// HTTPRoutes and GRPCRoute as target reference are extended
+	// support level.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+
+    // Type defines the method to achieve session persistence.
+	//
+	// Support: Core
+	//
+	// +required
+    Type *SessionPersistenceType `json:"type"`
+
+    // Duration defines the duration of the persistent session.
+    //
+    // Support: Core
+    //
+    // +optional
+    Duration int64 `json:"duration,omitempty"`
```

Comment:
There is one more "duration" that I just realized is very important: session draining duration (or timeout) which is described at https://github.com/grpc/proposal/blob/master/A55-xds-stateful-session-affinity.md#background (treat "affinity" as "persistence" in this linked doc). Basically a backend can get into "TERMINATING" state where it does not get any new sessions but it only stays around until its assigned sessions have terminated or the session-timeout triggers. I am wondering if this "session draining timeout" belongs in this policy - in what way does it apply to a gateway (or a mesh) to apply this timeout to the session persistence policy? One way I can think of is that let's say the "session draining timeout" is 1200seconds (20 mins). When a backend goes from HEALTHY to TERMINATING (or DRAINING) state the gateway will:

- the gateway will not pick that backend for sending new sessions
-  continue to send existing session requests to that backend for a maximum of 20 mins (unless the backend becomes UNHEALTHY or disappears before that).

What do you think? Sorry for bringing this up so late. We can always add this later as another PR.